lib/zosAccessor.js

```diff
@@ -411,42 +411,59 @@ ZosAccessor.prototype.rename = function (oldDataset, newDataset) {
 //==================================JES related operations=====================
 
 /**
- * List JES jobs matching query condition
+ * List JES jobs matching the given jobName or query option. The following parameters are accepted:
+ *
+ * @param {string} jobName - specify a JES job name, it can contain a wildcard (*)
  *
- * condition - An object whose valid keys are:
- *  jobName: specify a JES job name, it can contain a wildcard (*)
- *  owner:   specify a JES job owner, it can contain a wildcard (*)
- *  status:  specify a JES job status, eg. ALL, OUTPUT
+ * or
+ *
+ * @param {object} option - Option which contains
+ *    jobName: specify a JES job name, which is optional and can contain a wildcard (*)
+ *    jobId:   specify a JES job ID, which is optional
+ *    owner:   specify a JES job owner, which is optional and can contain a wildcard (*)
+ *    status:  specify a JES job status, eg. ALL, OUTPUT, which is optional
  */
-ZosAccessor.prototype.listJobs = function (condition) {
+ZosAccessor.prototype.listJobs = function (jobNameOrOption) {
     var deferred = Q.defer();
     var ftpClient = this.client;
+
+    var option = {};
+    if (typeof jobNameOrOption === 'string' || jobNameOrOption === undefined) {
+        // Default jobNameOrOption is job name for compatibility.
+        option = {jobName: jobNameOrOption || '*', owner: this.username, status: '*'};
+    } else if (typeof jobNameOrOption === 'object') {
+        option.jobName = jobNameOrOption.jobName || '*';
+        option.owner = jobNameOrOption.owner || this.username;
+        option.status = jobNameOrOption.status || '*';
+        if (jobNameOrOption.jobId) {
+            option.jobId = jobNameOrOption.jobId;
+        }
+    } else {
+        throw new Error('The first parameter type of listJobs() should be string or object.')
+    }
+
     ftpClient.ascii(function (err) {
         if (err) {
             return deferred.reject(err);
         }
-        if (typeof condition === 'string') {
-            // Default condition is job name for compatibility.
-            condition = {jobName: condition};
-        }
         var siteParams = ['FILETYPE=JES'];
-        for(var key in condition) {
-            if (!condition.hasOwnProperty(key)) {
+        for(var key in option) {
+            if (!option.hasOwnProperty(key)) {
```

Comment:
It is unnecessary to add this check `option.hasOwnProperty(key)`, right?

Reply:
`option.hasOwnProperty(key)` looks unnecessary here. It's kind of best practice to check whether `key` is owned by `option` object or inherited.